### DISCUSSION TITLE
Ability to Import/Export .mcstruture/.nbt files

### DISCUSSION THREAD

**Discussion by ihategravel2** [Mojang/minecraft-editor#26] [Feedback]
A great addition would to Editor Mode would be the ability to Import/Export .mcstruture/.nbt files. This would allow creators to save and import structures as needed. Creators could even import the vanilla .nbt structures if they have acess to them.

**Comment by MRBBATES1** [comment]
I agree, this would be major feature, to be able to bring in structures and preview where they are going before committing abit like paste preview would be great

**Comment by t3hero** [comment]
Drag and dropping structures directly into the client would be great as well!

**Comment by Landy005** [comment]
Would in addition to that .schem (schematics) files also work is that exclusively java only? Would be awesome though!

**Reply by ihategravel2** [reply]
Aren't schematic files a 3rd party file format? I don't think they would support that.

**Reply by Landy005** [reply]
True but dont think mcstructures are so globally used and integrated in every program as schematics? Like schematic to obj or obj to schem, fawe and other tool plugins, worldpainter and just java filesharing overall.

**Comment by minecraftchloe** [comment]
At risk of making a promise (😬) we have scoped out something like a "global asset/structure manager" for the Editor that would unlock a lot of this in tandem with some clipboard and general visualization improvements for paste preview we have planned. 

Schematics export - that is intriguing!
